services/authService.js

```diff
@@ -1,9 +1,44 @@
+import {
+  getUserSession
+} from '../sagas/apiCalls';
+
 let userInfoCache = {};
 
+export const openAuthorizationPopupWindow = (url) => {
+  const popup = window.open(url, url, 'toolbar=no,location=no,status=no,menubar=no,scrollbars=no,resizable=no,width=800,height=600');
+  setTimeout(() => {
+    if(!popup || popup.outerHeight === 0) {
+      alert("Popup Blocker is enabled! Please add this site to your exception list and try again.");
+    }
+  }, 25);
+};
+
 export const cacheUserInfo = (userInfo) => {
   userInfoCache = Object.assign({}, userInfo);
 }
 
 export const getUserJWT = () => {
-  return userInfoCache.jwt;
-}
\ No newline at end of file
+  return Promise.resolve(userInfoCache.jwt);
+}
+
+export const authorizeUser = (integrationUrl) => {
+  getUserJWT()
+  .then((jwt) => {
+    getUserSession(integrationUrl, jwt)
+    .then((userSession) => {
```

Comment:
Avoid highly nested Promises. As there's no error handling in the internal ones, just one level can be used to improve readability

Reply:
As we have seen, this is not necessary.